FundTrack.BLL/Concrete/FinOpService.cs

```diff
@@ -157,64 +157,142 @@ public FinOpViewModel CreateTransfer(FinOpViewModel finOpModel)
+                unitOfWork.FinOpRepository.Create(finOp);
+                orgAccFrom.CurrentBalance -= finOpModel.Amount;
+                unitOfWork.OrganizationAccountRepository.Edit(orgAccFrom);
+                orgAccTo.CurrentBalance += finOpModel.Amount;
+                unitOfWork.OrganizationAccountRepository.Edit(orgAccTo);
+                unitOfWork.SaveChanges();
                 return finOpModel;
             }
             catch (Exception ex)
             {
                 throw new BusinessLogicException("Щось пішло не так....О_о", ex);
             }
         }
-        /// <summary>
+        /// <Amountmary>
         /// Gets the fin ops by org account.
-        /// </summary>
+        /// </Amountmary>
         /// <param name="orgAccountId">The org account identifier.</param>
         /// <returns></returns>
-        public IEnumerable<FinOpListViewModel> GetFinOpsByOrgAccount(int orgAccountId)
+        public IEnumerable<FinOpViewModel> GetFinOpsByOrgAccount(int orgAccountId)
         {
             try
             {
-                var finOps = _unitOfWork.FinOpRepository.GetFinOpByOrgAccount(orgAccountId)
+                var finOps = unitOfWork.FinOpRepository.GetFinOpByOrgAccount(orgAccountId)
                     .OrderByDescending(f => f.Id)
-                    .Select(f => new FinOpListViewModel
+                    .Select(f => new FinOpViewModel
                     {
+                        Id = f.Id,
+                        CardFromId = f.AccFromId.GetValueOrDefault(0),
+                        CardToId = f.AccToId.GetValueOrDefault(0),
                         Date = f.FinOpDate,
                         Description = f.Description,
                         Amount = f.Amount,
+                        TargetId = f.TargetId,
                         Target = f.Target.TargetName,
-                        CurrencyShortName = f.AccToId.HasValue ? f.OrgAccountTo.Currency.ShortName : f.OrgAccountFrom.Currency.ShortName,
-                        CurrencyFullName = f.AccToId.HasValue ? f.OrgAccountTo.Currency.FullName : f.OrgAccountFrom.Currency.FullName,
-                        FinOpType = f.FinOpType
+                        FinOpType = f.FinOpType,
+                        IsEditable = true
                     });
                 return finOps;
             }
             catch (Exception ex)
             {
-                return new FinOpListViewModel
+                return new FinOpViewModel
+                {
+                    Error = "Список фінансових операцій порожній."
+                } as IEnumerable<FinOpViewModel>;
+            }
+        }
+        /// <Amountmary>
+        /// Gets the fin ops by id.
+        /// </Amountmary>
+        /// <param name="id">The fin ops identifier.</param>
+        /// <returns></returns>
+        public FinOpViewModel GetFinOpsById(int id)
+        {
+            try
+            {
+                var f = unitOfWork.FinOpRepository.GetById(id);

```

Comment:
naming!

Reply:
agree, give this variable more meaningfull name.